py_gtfs_rt_ingestion/lib/convert_gtfs_rt.py

```diff
@@ -186,6 +193,39 @@ def gz_to_pyarrow(self, filename: str) -> Tuple[datetime, pyarrow.table]:
             pyarrow.table(table, schema=self.detail.export_schema),
         )
 
-    @property
-    def partition_cols(self) -> list[str]:
-        return ["year", "month", "day", "hour"]
+    def write_and_archive(self, table: pyarrow.table) -> None:
+        """
+        write the table to our s3 bucket and move archive and error files to
+        their respective s3 buckets.
+        """
+        try:
+            s3_prefix = str(self.config_type)
+            write_parquet_file(
+                table=table,
+                filetype=s3_prefix,
+                s3_path=os.path.join(
+                    os.environ["SPRINGBOARD_BUCKET"],
+                    DEFAULT_S3_PREFIX,
+                    s3_prefix,
+                ),
+                partition_cols=["year", "month", "day", "hour"],
+            )
+
+        except Exception:
+            self.error_files += self.archive_files
+            self.archive_files = []
+
+        finally:
+            if len(self.error_files) > 0:
+                move_s3_objects(
+                    self.error_files,
+                    os.path.join(os.environ["ERROR_BUCKET"], DEFAULT_S3_PREFIX),
+                )
+
+            if len(self.archive_files) > 0:
+                move_s3_objects(
+                    self.archive_files,
+                    os.path.join(
+                        os.environ["ARCHIVE_BUCKET"], DEFAULT_S3_PREFIX
+                    ),
+                )
```

Comment:
pretty straight forward. this was what we were doing in the ingest script before.